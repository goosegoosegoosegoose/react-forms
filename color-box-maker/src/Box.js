import React from "react";

const Box = ({width, height, bgcolor, remove}) => {
    return (
        <div>
            <div className="box" style={{backgroundColor: `${bgcolor}`,
                                        width: `${width}rem`, 
                                        height: `${height}rem`,
                                        display: "inline-block"}} >
                <button onClick={remove}>X</button>
            </div>
        </div>
    )
}

export default Box;