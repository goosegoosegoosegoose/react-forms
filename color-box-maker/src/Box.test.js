import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box"

it("smoke test", () => {
    render(<Box />);
});

it("snapshot test", () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
})