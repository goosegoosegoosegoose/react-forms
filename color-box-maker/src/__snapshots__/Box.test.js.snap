// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<DocumentFragment>
  <div
    class="box"
    style="display: inline-block;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
